### PR TITLE
upstream lb: Improve config check.

### PR DESCRIPTION
**Pull request by vehre-x41**
Commit Message: upstream lb: Improve config check.
Additional Description:
Prevent illegal config value for aggression_runtime to set
aggression_ to nan.
Signed-off-by: Andre Vehreschild <vehre@x41-dsec.de>
Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by vehre-x41** [comment]
CC @adisuissa @KBaichoo @pradeepcrao

**Review by KBaichoo — APPROVED** [review]
lgtm

File: source/common/upstream/load_balancer_impl.cc [hunk 1]
```diff
@@ -997,11 +997,11 @@ double EdfLoadBalancerBase::applySlowStartFactor(double host_weight, const Host&
   if (host_create_duration < slow_start_window_ &&
       host.health() == Upstream::Host::Health::Healthy) {
     aggression_ = aggression_runtime_ != absl::nullopt ? aggression_runtime_.value().value() : 1.0;
-    if (aggression_ < 0.0) {
+    if (aggression_ <= 0.0 || std::isnan(aggression_)) {
       ENVOY_LOG_EVERY_POW_2(error, "Invalid runtime value provided for aggression parameter, "
                                    "aggression cannot be less than 0.0");
+      aggression_ = 1.0;
```
Comment (adisuissa) [inline source/common/upstream/load_balancer_impl.cc:1003]: @nezdolik Is this a good value if the set aggression value is invalid?

Comment (KBaichoo) [reply source/common/upstream/load_balancer_impl.cc:1003]: that's the default value if this is unset so it seemed reasonable.

**Pull request merged**